comp/metadata/inventoryotel/inventoryotelimpl/inventoryotel.go

```diff
@@ -0,0 +1,270 @@
+type Payload struct {
+	Hostname  string       `json:"hostname"`
+	Timestamp int64        `json:"timestamp"`
+	Metadata  otelMetadata `json:"otel_metadata"`
+	UUID      string       `json:"uuid"`
+}
+
+// MarshalJSON serialization a Payload to JSON
+func (p *Payload) MarshalJSON() ([]byte, error) {
+	type PayloadAlias Payload
+	return json.Marshal((*PayloadAlias)(p))
+}
+
+// SplitPayload implements marshaler.AbstractMarshaler#SplitPayload.
+//
+// In this case, the payload can't be split any further.
+func (p *Payload) SplitPayload(_ int) ([]marshaler.AbstractMarshaler, error) {
+	return nil, fmt.Errorf("could not split inventories agent payload any more, payload is too big for intake")
+}
+
+type inventoryotel struct {
+	util.InventoryPayload
+
+	conf      config.Component
+	log       log.Component
+	m         sync.Mutex
+	data      otelMetadata
+	hostname  string
+	authToken authtoken.Component
+	f         *freshConfig
+}
+
+type dependencies struct {
+	fx.In
+
+	Log        log.Component
+	Config     config.Component
+	Serializer serializer.MetricSerializer
+	AuthToken  authtoken.Component
+}
+
+type provides struct {
+	fx.Out
+
+	Comp                 iointerface.Component
+	Provider             runnerimpl.Provider
+	FlareProvider        flaretypes.Provider
+	StatusHeaderProvider status.HeaderInformationProvider
+	Endpoint             api.AgentEndpointProvider
+}
+
+func newInventoryOtelProvider(deps dependencies) provides {
+	hname, _ := hostname.Get(context.Background())
+	i := &inventoryotel{
+		conf:      deps.Config,
+		log:       deps.Log,
+		hostname:  hname,
+		data:      make(otelMetadata),
+		authToken: deps.AuthToken,
+	}
+
+	getter := i.fetchRemoteOtelConfig
+	if i.conf.GetBool("otel.submit_dummy_inventories_metadata") {
+		getter = i.fetchDummyOtelConfig
+	}
+
+	var err error
+	i.f, err = newFreshConfig(deps.Config.GetString("otel.extension_url"), getter)
+	if err != nil {
+		// panic?
+		return provides{}
+	}
+
+	i.InventoryPayload = util.CreateInventoryPayload(deps.Config, deps.Log, deps.Serializer, i.getPayload, "otel.json")
+
+	if i.Enabled {
+
+		// TODO: if there's an update on the OTel side we currently will not be
+		//       notified. Is this a problem? Runtime changes are expected to be
+		//       triggered by FA, so maybe this is OK.
+		//
+		// We want to be notified when the configuration is updated
+		deps.Config.OnUpdate(func(_ string, _, _ any) { i.Refresh() })
+	}
+
+	return provides{
+		Comp:                 i,
+		Provider:             i.MetadataProvider(),
+		FlareProvider:        i.FlareProvider(),
+		StatusHeaderProvider: status.NewHeaderInformationProvider(i),
+		Endpoint:             api.NewAgentEndpointProvider(i.writePayloadAsJSON, "/metadata/inventory-otel", "GET"),
+	}
+}
+
+func scrub(s string) string {
+	// Errors come from internal use of a Reader interface. Since we are reading from a buffer, no errors
+	// are possible.
+	scrubString, _ := scrubber.ScrubString(s)
+	return scrubString
+}
```

Comment:
Should we move this function to `comp/metadata/inventoryotel/inventoryotelimpl/configuration.go` as it is the only place is used? We could even inline the code inside `copyAndScrub` function